lib/screens/stake_screen/stake_screen.dart

```diff
@@ -0,0 +1,340 @@
+  pendingApprove,
+  canStake,
+  stakePendingApprove,
+  stakeApproved
+}
+
+class StakeScreen extends StatefulWidget {
+  static const url = '/stake';
+
+  @override
+  _StakeScreenState createState() => _StakeScreenState();
+}
+
+class _StakeScreenState extends State<StakeScreen> {
+  final String address = '0x3a6dabD6B5C75291a3258C29B418f5805792a87e';
+
+  final double apy = 99.93;
+
+  final balance = 13.234;
+
+  final gradient = MyColors.blueToGreenGradient;
+
+  StakesStates _stakeState = StakesStates.hasToApprove;
+  bool _showToast = false;
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Color(MyColors.Background),
+      body: SafeArea(
+        child: SingleChildScrollView(
+          padding: const EdgeInsets.symmetric(vertical: 20, horizontal: 15),
+          child: SizedBox(
+            height: MediaQuery.of(context).size.height-80,
+            child: Column(
+              mainAxisSize: MainAxisSize.min,
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                _buildHeaderAddress(),
+                SizedBox(
+                  height: 23,
+                ),
+                _buildBackButton(),
+                SizedBox(
+                  height: 20,
+                ),
+                Text(
+                  'Stake your sDEA',
+                  style: TextStyle(fontSize: 25),
+                ),
+                SizedBox(
+                  height: 8,
+                ),
+                Text(
+                  '$apy% APY',
+                  style: TextStyle(fontSize: 20),
+                ),
+                SizedBox(
+                  height: 20,
+                ),
+                Padding(
+                  padding: const EdgeInsets.only(left: 15),
+                  child: Text(
+                    'Balance: $balance',
+                    style: MyStyles.lightWhiteSmallTextStyle,
+                  ),
+                ),
+                SizedBox(
+                  height: 5,
+                ),
+                _buildTextField(),
+                SizedBox(
+                  height: 12,
+                ),
+                DarkButton(
+                  label: 'Show me the contract',
+                  onPressed: () {},
+                  labelStyle: MyStyles.whiteMediumTextStyle,
+                ),
+                SizedBox(
+                  height: 12,
+                ),
+                _buildStakeApproveButton(),
+                SizedBox(
+                  height: 15,
+                ),
+                if (_stakeState == StakesStates.hasToApprove)
+                  _buildSteps(),
+                Spacer(),
+                if ((_stakeState != StakesStates.hasToApprove && _showToast) || (_stakeState == StakesStates.stakePendingApprove && _showToast))
+                  _buildToast()
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Container _buildToast() {
```

Comment:
Der Code ist insgesamt zwar ok, aber ich würde versuchen das stärker zu trennen. Wenn wir vom Backend eine Antwort kriegen, ob eine Transaktion erfolgreich war, möchten wir die ja nicht erst an eine buildToast Methode schicken, ohne zu wissen, was wir danach rauskriegen. Wir haben ja im Regelfall eh schon die Werte, ob die Transaktion geklappt hat oder nicht. Also würde ich diese einzelne Funktion in zwei Funktionen aufteilen, eine für den erfolgreichen Fall und eine für pending.

Da beide Toasts vermutlich das gleiche Grundschema behalten, kannst du auch eine _buildToast Methode schreiben. Noch besser wäre vermutlich sogar ein extra Widget dafür, weil du ja im Lockscreen nahezu das gleiche brauchst, oder?

- [ ] Funktion _buildToast(String label, Color color, {VoidCallback onPressed}) ODER ein Widget dafür
- [ ] _buildTransactionSuccessToast
- [ ] _buildTransactionPendingToast
- [ ] Conditions oben einfügen und dann je nach Fall success bzw. pendingToast anzeigen